Mergin/projects_manager.py

```diff
@@ -234,26 +266,23 @@ def sync_project(self, project_dir, project_name=None):
                 )
             return
 
+        # after pull project might be in the unfinished pull state. So we
+        # have to check and if this is the case, try to close project and
+        # finish pull. As in the result we will have conflicted copies created
+        # we stop and ask user to examine them.
+        if self.mc.has_unfinished_pull(project_dir):
+            current_project_path = os.path.normpath(QgsProject.instance().absolutePath())
+            delay = 0
+            if current_project_path == os.path.normpath(project_dir):
+                QgsProject.instance().clear()
+                delay = 2500
+            # we have to wait a bit to let the OS (Windows) release lock on the GPKG files
+            # otherwise attempt to resolve unfinished pull will fail
+            QTimer.singleShot(delay, lambda: self.resolve_unfinished_pull(project_dir, True))
```

Comment:
as this code to close project + deferred resolve is present three times, it may be good to move it to a separate function, so that the logic is in one place? (especially if we ever need to modify the delay time or change the logic to retry few times)